components/wso2is.notification.event.handlers/src/main/java/org/wso2/is/notification/ApimOauthEventInterceptor.java

```diff
@@ -0,0 +1,127 @@
+                                              RefreshTokenValidationDataDO refreshTokenDO, Map<String, Object> params)
+            throws IdentityOAuth2Exception {
+
+        if (enabled && accessTokenDO != null) {
+            try {
+                long expiryTime = accessTokenDO.getIssuedTime().getTime() + accessTokenDO.getValidityPeriodInMillis();
+                String accessToken = accessTokenDO.getAccessToken();
+                String user = accessTokenDO.getAuthzUser().getUserName();
+                int tenantID = accessTokenDO.getTenantID();
+                String tenantDomain =
+                        ServiceReferenceHolder.getInstance().getRealmService().getTenantManager().getDomain(tenantID);
+                OAuthAppDO oauthApp = OAuth2Util.getAppInformationByClientId(accessTokenDO.getConsumerKey());
+                String tokenType = oauthApp.getTokenType();
+                TokenRevocationEvent tokenRevocationEvent = new TokenRevocationEvent(accessToken, expiryTime, user,
+                        accessTokenDO.getConsumerKey(), tokenType);
+                tokenRevocationEvent.setTenantId(tenantID);
+                tokenRevocationEvent.setTenantDomain(tenantDomain);
+                publishEvent(tokenRevocationEvent);
+            } catch (InvalidOAuthClientException e) {
+                log.error("Error while retrieving token type", e);
+            } catch (UserStoreException e) {
+                log.error("Error while resolving tenantDomain", e);
+            }
+        }
+
+    }
+
+    @Override
+    public void onPostTokenRevocationByResourceOwner(
+            org.wso2.carbon.identity.oauth.dto.OAuthRevocationRequestDTO revokeRequestDTO,
+            org.wso2.carbon.identity.oauth.dto.OAuthRevocationResponseDTO revokeRespDTO,
+            AccessTokenDO accessTokenDO, Map<String, Object> params)
+            throws IdentityOAuth2Exception {
+
+        if (enabled && accessTokenDO != null) {
+            try {
+                long expiryTime = accessTokenDO.getIssuedTime().getTime() + accessTokenDO.getValidityPeriodInMillis();
+                String accessToken = accessTokenDO.getAccessToken();
+                String user = accessTokenDO.getAuthzUser().getUserName();
+                OAuthAppDO oauthApp = OAuth2Util.getAppInformationByClientId(accessTokenDO.getConsumerKey());
+                String tokenType = oauthApp.getTokenType();
+                TokenRevocationEvent tokenRevocationEvent = new TokenRevocationEvent(accessToken, expiryTime, user,
+                        accessTokenDO.getConsumerKey(), tokenType);
+                publishEvent(tokenRevocationEvent);
+            } catch (InvalidOAuthClientException e) {
+                log.error("Error while retrieving token type", e);
+            }
+        }
+    }
+
+    private void publishEvent(TokenRevocationEvent tokenRevocationEvent) {
+
+        if (enabled && isEnabled()) {
+            if (StringUtils.isNotEmpty(notificationEndpoint)) {
```

Comment:
Isn't this redundant? State of enabled is set after checking notificationEndpoint. Why do we need to check it again?